folium/plugins/overlapping_marker_spiderfier.py

```diff
@@ -0,0 +1,86 @@
+from jinja2 import Template
+
+from folium.elements import JSCSSMixin, MacroElement
+from folium.utilities import parse_options
+
+
+class OverlappingMarkerSpiderfier(JSCSSMixin, MacroElement):
+    """
+    A plugin that handles overlapping markers on a map by spreading them out in a spiral or circle pattern when clicked.
+
+    This plugin is useful when you have multiple markers in close proximity that would otherwise be difficult to interact with.
+    When a user clicks on a cluster of overlapping markers, they spread out in a 'spider' pattern, making each marker
+    individually accessible.
+
+    Markers must be added to the map **before** calling `oms.add_to(map)`.
+    The plugin identifies and manages all markers already present on the map.
+
+    Parameters
+    ----------
+    options : dict, optional
+        The options to configure the spiderfier behavior:
+        - keepSpiderfied : bool, default True
+            If true, markers stay spiderfied after clicking
+        - nearbyDistance : int, default 20
+            Pixels away from a marker that is considered overlapping
+        - legWeight : float, default 1.5
+            Weight of the spider legs
+        - circleSpiralSwitchover : int, optional
+            Number of markers at which to switch from circle to spiral pattern
+
+    Example
+    -------
+    >>> oms = OverlappingMarkerSpiderfier(
+    ...     options={"keepSpiderfied": True, "nearbyDistance": 30, "legWeight": 2.0}
+    ... )
+    >>> oms.add_to(map)
+    """
+
+    _template = Template(
+        """
+        {% macro script(this, kwargs) %}
+        (function () {
+            try {
+                var oms = new OverlappingMarkerSpiderfier(
+                    {{ this._parent.get_name() }},
+                    {{ this.options|tojson }}
+                );
+
+                oms.addListener('spiderfy', function() {
+                    {{ this._parent.get_name() }}.closePopup();
+                });
+
+                {{ this._parent.get_name() }}.eachLayer(function(layer) {
+                    if (
+                        layer instanceof L.Marker
+                    ) {
+                        oms.addMarker(layer);
+                    }
+                });
```

Comment:
This works great when users add their markers directly to the map object. But it's quite common to add markers to a FeatureGroup, then add the FeatureGroup to the map. I think we should support that use case.

This made me think whether it's maybe better to explicitly pass the markers they want to have spiderified like before. That way, they can group them in FeatureGroups however they wish. Downside is it doesn't work automatically like here. 

Like for example

```suggestion
                {%- for marker in this.markers %}
                    oms.addMarker({{ marker.get_name() }});
                {%- endfor %}
```

Alternatively, expand this algorithm to recursively go through all layers. For example something like

```
let markers = [];

function findMarkers(layer) {
    if (layer instanceof L.Marker) {
        markers.push(layer);
    } else if (layer instanceof L.LayerGroup) {
        layer.eachLayer(findMarkers); // Recursively check nested layers
    }
}

map.eachLayer(findMarkers);
```

On the one hand making things easier for users is good. On the other hand, it's more code and functionality that can have bugs or can break.

What do you think?